products/bash/setup-demos.sh

```diff
@@ -161,46 +177,90 @@ function merge_addon() {
   # add condition to condition array
   STATUS=$(echo $STATUS | jq -c '.conditions += ['"${CONDITION_TO_ADD}"']')
-  $DEBUG && echo -e "\n$info [INFO] Printing the status conditions array" && echo $STATUS | jq -r '.conditions'
+  $DEBUG && echo -e "\n$info [DEBUG] Printing the status conditions array" && echo $STATUS | jq -r '.conditions'
 }
 
+#----------------------------------------------------
+
 function update_phase() {
   PHASE=${1} # Pending, Running or Failed
-
-  $DEBUG && divider && echo -e "$info [INFO] update_phase(): phase($PHASE)"
-
+  $DEBUG && divider && echo -e "$info [DEBUG] update_phase(): phase($PHASE)"
   STATUS=$(echo $STATUS | jq -c '.phase="'$PHASE'"')
 }
 
-function check_phase_and_exit_on_failed() {
+#----------------------------------------------------
 
+function check_phase_and_exit_on_failed() {
   CURRENT_PHASE=$(echo $STATUS | jq -r '.phase')
-
   # if the current phase is failed, then exit status (case insensitive checking)
   if echo $CURRENT_PHASE | grep -iqF failed; then
     divider && echo -e "$info [INFO] Current installation phase is '$CURRENT_PHASE', exiting now." && divider
     exit 1
   else
-    $DEBUG && divider && echo -e "$info [INFO] Current installation phase is '$CURRENT_PHASE', continuing the installation..."
+    $DEBUG && divider && echo -e "$info [DEBUG] Current installation phase is '$CURRENT_PHASE', continuing the installation..."
   fi
 }
 
+#----------------------------------------------------
+
+function update_addon_status() {
+  ADDON_TYPE=${1}         # type of addon
+  ADDON_INSTALLED=${2}    # if the addon is installed
+  ADDON_READY_TO_USE=${3} # if the installed addon is configured and ready to use
+
+  $DEBUG && divider && echo -e "$info [DEBUG] addonType($ADDON_TYPE) - addonInstalled($ADDON_INSTALLED) - addonReadyToUse($ADDON_READY_TO_USE)"
+
+  # clear any existing status for the passed addon type
+  STATUS=$(echo $STATUS | jq -c 'del(.addons[] | select(.type == "'$ADDON_TYPE'")) ')
+  # create object and add status for each addon
+  ADDON_TO_ADD_TO_STATUS=$(echo $ADDON_OBJECT_FOR_STATUS | jq -r '.type="'$ADDON_TYPE'" | .installed="'$ADDON_INSTALLED'" | .readyToUse="'$ADDON_READY_TO_USE'" ')
+  # update status with new addon status
+  STATUS=$(echo $STATUS | jq -c '.addons += ['"${ADDON_TO_ADD_TO_STATUS}"']')
+}
+
+#----------------------------------------------------
+
+function update_product_status() {
+  PRODUCT_NAME=${1}         # name of product
+  PRODUCT_TYPE=${2}         # type of product
+  PRODUCT_INSTALLED=${3}    # if the product is installed
+  PRODUCT_READY_TO_USE=${4} # if the installed product is configured and ready to use
+
+  $DEBUG && divider && echo -e "$info [DEBUG] productName($PRODUCT_NAME) - productType($PRODUCT_TYPE) - productInstalled($PRODUCT_INSTALLED) - productReadyToUse($PRODUCT_READY_TO_USE)"
+
+  # clear any existing status for the passed product type
+  STATUS=$(echo $STATUS | jq -c 'del(.products[] | select(.type == "'$PRODUCT_TYPE'")) ')
```

Comment:
We're not currently using the feature, but it's possible to have multiple products of the same type but with different names, and also different namespaces. I think change this to match the predicate used in `merge_product`:
```
MATCH=$(echo "${PRODUCT_ARRAY_JSON}" | jq -c '.[] | select(.type == "'$TYPE'" and .name == "'$NAME'" and .namespace == "'$NAMESPACE'")')
```

Reply:
Done